### PR TITLE
Update day 2-4 to latest version of hypersync

### PR DESCRIPTION
**Pull request by JasoonS**
I'm noting that there is a ~4x speed improvement when moving requests to the backup cluster.
I left the links pointing to the main cluster so that we don't need to change things once backup cluster is migrated over.

### CODE REVIEW

File: 1-fetch-all-logs/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 **Prerequisites**
 
-- Install `hypersync` and `asyncio` libraries using uv or pip: `uv pip install hypersync asyncio`
```
Comment (JasoonS) [inline 1-fetch-all-logs/README.md:5]: No point having a requirements.txt file if you don't use it!

File: 1-fetch-all-logs/requirements.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # This file was autogenerated by uv via the following command:
 #    uv pip compile - -o requirements.txt
 asyncio==3.4.3
-hypersync==0.6.2
+hypersync==0.7.1
```
Comment (JasoonS) [inline 1-fetch-all-logs/requirements.txt:4]: Did this all around.

File: 1-fetch-all-logs/fetch-logs.py
Comment (JasoonS) [inline 1-fetch-all-logs/fetch-logs.py:1]: Uses latest client api

File: 2-usdc-transfer-volume/usdc-volume.py [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
-from hypersync import LogSelection, LogField, FieldSelection, TransactionField, ColumnMapping, DataType
+from hypersync import LogSelection, FieldSelection, LogField, TransactionField, DataType, ColumnMapping;
 import asyncio
 
 async def collect_events():
     # Choose network
-    client = hypersync.HypersyncClient("https://eth.hypersync.xyz")
+    client = hypersync.HypersyncClient(hypersync.ClientConfig(url="https://eth.hypersync.xyz"))
 
+    print("constructing query...")
```
Comment (JasoonS) [inline 2-usdc-transfer-volume/usdc-volume.py]:
```suggestion
```

My bad, this was just debugging...

File: 2-usdc-transfer-volume/usdc-volume.py [hunk 4]
```diff
@@ -29,11 +30,9 @@ async def collect_events():
         ),
     )
 
-    config = hypersync.ParquetConfig(
-        path="data",
-        hex_output=True, 
-        batch_size=50000,
-        concurrency=12,
+    print("parquet config...")
```
Comment (JasoonS) [inline 2-usdc-transfer-volume/usdc-volume.py]:
```suggestion
```

File: 2-usdc-transfer-volume/usdc-volume.py [hunk 5]
```diff
@@ -43,10 +42,11 @@ async def collect_events():
         event_signature="Transfer(address indexed from, address indexed to, uint256 value)",
     )
 
-    await client.create_parquet_folder(query, config)
+    print("executing query...")
```
Comment (JasoonS) [inline 2-usdc-transfer-volume/usdc-volume.py:43]:
```suggestion
```

File: 3-all-erc20-transfers-base/base-transfers.py [hunk 6]
```diff
@@ -15,30 +15,27 @@ async def collect_events():
...
-                LogField.DATA,
-                LogField.TRANSACTION_HASH,
+                # Loading only this simple boolean field makes this erquest much faster
+                LogField.REMOVED,
+                # LogField.TOPIC0,
+                # LogField.TOPIC1,
+                # LogField.TOPIC2,
+                # LogField.TOPIC3,
+                # LogField.DATA,
+                # LogField.TRANSACTION_HASH,
```
Comment (JasoonS) [inline 3-all-erc20-transfers-base/base-transfers.py]: Should we just delete this all?
```suggestion
```

File: 3-all-erc20-transfers-base/requirements.txt [hunk 7]
```diff
@@ -1,7 +1,9 @@
 # This file was autogenerated by uv via the following command:
 #    uv pip compile - -o requirements.txt
 asyncio==3.4.3
-hypersync==0.6.2
-polars==0.20.25
```
Comment (JasoonS) [inline 3-all-erc20-transfers-base/requirements.txt:5]: Polars isn't needed in this example.

File: 4-transaction-to-and-from-address/requirements.txt [hunk 8]
```diff
@@ -1,7 +1,6 @@
 # This file was autogenerated by uv via the following command:
 #    uv pip compile - -o requirements.txt
 asyncio==3.4.3
-hypersync==0.6.2
-polars==0.20.26
```
Comment (JasoonS) [inline 4-transaction-to-and-from-address/requirements.txt:5]: Not used once again.

**Review by moose-code — APPROVED** [review]
Looks good!!

**Pull request merged**